src/lava/lib/optimization/apps/scheduler/solver.py

```diff
@@ -0,0 +1,289 @@
+        hp_update (tuple of dict and bool) :
+        The bool part toggles whether to update the existing hyperparameters
+            or to set new ones from scratch.
+
+        The dict part of the tuple has the following keys:
+            - temperature: annealing parameter. Value of this key should be
+            an integer, typically between 1 and 10. Default is 8.
+            - refract: refractory period for each neuron for which it stays
+            "on" or "off". Value of this key should be a NumPy integer array of
+            length equal to the number of nodes in the problem graph. Default
+            value is comprised by random integers between 64 and 127.
+            - refract_counter: Counter for the *initial* number of time-steps
+            for which neurons should stay "off". Value of this key should be
+            a NumPy integer array of length equal to the number of nodes in
+            the problem graph. Default value is comprised by random integers
+            between 0 and 64.
+        """
+        update = hp_update[1]
+        if not update:
+            self._qubo_hyperparams = hp_update[0]
+        else:
+            self._qubo_hyperparams.update(hp_update[0])
+
+    @property
+    def qubo_weights(self):
+        return self._qubo_weights
+
+    @qubo_weights.setter
+    def qubo_weights(self, qw: Tuple[int, int]):
+        self._qubo_weights = qw
+
+    @property
+    def qubo_target_cost(self):
+        return self._qubo_target_cost
+
+    @property
+    def probe_cost(self):
+        return self._probe_cost
+
+    @probe_cost.setter
+    def probe_cost(self, val: bool):
+        """Toggle whether to probe cost during the solver run.
+        Parameters
+        ----------
+        val (bool) : Default is False.
+        """
+        self._probe_cost = val
+
+    @property
+    def probe_loihi_exec_time(self):
+        return self._probe_loihi_exec_time
+
+    @property
+    def probe_loihi_energy(self):
+        return self._probe_loihi_energy
+
+    @property
+    def netx_solution(self):
+        return self._netx_solution
+
+    @property
+    def qubo_problem(self):
+        return self._qubo_problem
+
+    @property
+    def qubo_matrix(self):
+        return self._qubo_matrix
+
+    @property
+    def lava_backend(self):
+        return self._lava_backend
+
+    @lava_backend.setter
+    def lava_backend(self, backend: str):
+        self._lava_backend = backend
+
+    @property
+    def lava_solver_report(self):
+        return self._lava_solver_report
+
+    @property
+    def lava_solution(self):
+        return self._lava_solution
+
+    def gen_qubo_mat(self):
+        adj_mat = self.problem.adjacency
+        self._qubo_matrix = MISProblem._get_qubo_cost_from_adjacency(
+            adj_mat, self.qubo_weights[0], self.qubo_weights[1])
+
+    def gen_qubo_problem(self):
+        self.gen_qubo_mat()
+        self._qubo_problem = QUBO(self.qubo_matrix)
+
+    def solve_with_netx(self):
```

Comment:
Could you add a few comments about what is happening here?

Reply:
Done.